src/xma/include/lib/xmahw_lib.h

```diff
@@ -61,6 +61,17 @@ constexpr std::uint64_t signature = 0xF42F1F8F4F2F1F0F;
 /* Forward declaration */
 typedef struct XmaHwDevice XmaHwDevice;
 
+enum class xma_cmd_state_priv: std::int32_t {
+  QUEUED = 1, //Submitted to XMA -> XRT
+  COMPLETED = 2, //Cmd has finished
+  ERROR = 3, //XMA or XRT error during submission of cmd
+  ABORT = 4, //XRT aborted the cmd; CU may or may not have received the cmd
+  TIMEOUT = 5, //XMA or XRT timeout waiting for cmd to finish
+  PSK_ERROR = 6, //PS Kernel cmd completed but with error return code
+  PSK_CRASHED = 7, //PS kernel has crashed
+  MAX = 8 // Always the last one
```

Comment:
It's kind of magic that the numbers must match the external classic enum.

You should (1) use lower case enumerators, otherwise you have the same problem as to begin with in that some braindead code could #define MAX, ABORT, ERROR, ... to something.  (2) initialize the enumerators from corresponding external enumerators
```
queued = XMA_QUEUED,
completed = XMA_COMPLETED,
etc. 
```
When I look at the xmahw_lib.h external header file, it is using CamelCase for C facing types, which is good, so I think it should have used XmaCmdState to match the C naming convention (see XmaHwSession for example).  Your C++ facing enum class could then have used just xma_cmd_state.

The XRT naming convention is foo_underscore_bar for C++ types, functions, and variables, and CamelCase or C stuff; it looks like XMA agrees with this in general.

Reply:
Updated